cylc/flow/dbstatecheck.py

```diff
@@ -145,37 +180,49 @@ def workflow_state_query(
             stmt_wheres.append("cycle==?")
             stmt_args.append(cycle)
 
-        if status:
+        if status is not None:
             stmt_frags = []
             for state in self.state_lookup(status):
                 stmt_args.append(state)
                 stmt_frags.append("status==?")
             stmt_wheres.append("(" + (" OR ").join(stmt_frags) + ")")
+
         if stmt_wheres:
             stmt += " where " + (" AND ").join(stmt_wheres)
 
+        # Note we can't use "where outputs==message"; because the outputs
+        # table holds a serialized string of all received outputs.
+
         res = []
         for row in self.conn.execute(stmt, stmt_args):
-            if not all(v is None for v in row):
+            if row[-1] is not None:
+                # not all(v is None for v in row):
```

Comment:
Note to self: fix comment. I can't see how the original test makes sense, but it is possible for the task_outputs table (a Cylc 7 one at least) to have a row with outputs value None.